torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -250,3 +256,57 @@ def _make_sharded_datapipe(root: str, dataset_size: int) -> IterDataPipe:
     # dp = dp.cycle(2)
     dp = TakerDataPipe(dp, dataset_size)
     return dp
+
+
+def fromfile(
```

Comment:
It would be great to test this util a bit, e.g. against various equivalent parametrization of `numpy.fromfile`.

One important thing to test seems to be that we never get the "non-writable buffer" warning from pytorch.